packages/next/shared/lib/router/utils/format-next-pathname-info.ts

```diff
@@ -0,0 +1,28 @@
+import type { NextPathnameInfo } from './get-next-pathname-info'
+import { addPathPrefix } from './add-path-prefix'
+import { addPathSuffix } from './add-path-suffix'
+import { addLocale } from './add-locale'
+
+interface ExtendedInfo extends NextPathnameInfo {
+  defaultLocale?: string
+}
+
+export function formatNextPathnameInfo(info: ExtendedInfo) {
+  let pathname = addLocale(
+    info.pathname,
+    info.locale,
+    info.buildId ? undefined : info.defaultLocale
+  )
+
+  if (info.buildId) {
+    pathname = addPathSuffix(
+      addPathPrefix(pathname, `/_next/data/${info.buildId}`),
+      info.pathname === '/' ? 'index.json' : '.json'
```

Comment:
Is the locale already meant to be included here if present?

Reply:
The locale should be already part of `pathname` as it is added before. If there is no `buildId` it will be _always_ added